test/xUnit/csharp/test_CompletionHelpers.cs

```diff
@@ -8,6 +8,62 @@ namespace PSTests.Parallel
 {
     public class CompletionHelpersTests
     {
+        [Theory]
+        [InlineData("", "'", "''")]
+        [InlineData("", "\"", "\"\"")]
+        [InlineData("'", "'", "''''")]
+        [InlineData("", "", "''")]
+        [InlineData("", null, "''")]
+        [InlineData("word", "'", "'word'")]
+        [InlineData("word", "\"", "\"word\"")]
+        [InlineData("word's", "'", "'word''s'")]
+        [InlineData("it's a test", "'", "'it''s a test'")]
+        [InlineData("already 'quoted'", "'", "'already ''quoted'''")]
+        [InlineData("multiple 'quotes' in 'text'", "'", "'multiple ''quotes'' in ''text'''")]
+        [InlineData("\"word\"", "'", "'\"word\"'")]
+        [InlineData("'word'", "'", "''word''")]
+        [InlineData("word", "", "word")]
+        [InlineData("word", null, "word")]
+        [InlineData("\"word\"", "\"", "\"\"word\"\"")]
+        [InlineData("'word'", "\"", "\"'word'\"")]
+        [InlineData("word with space", "'", "'word with space'")]
+        [InlineData("word with space", "\"", "\"word with space\"")]
+        [InlineData("word\"with\"quotes", "'", "'word\"with\"quotes'")]
+        [InlineData("word'with'quotes", "\"", "\"word'with'quotes\"")]
```

Comment:
I'd add tests for "while", "$variable" and "key$word".

Reply:
@iSazonov I believe we have tests for those already?

Reply:
Or do we want the same test from CompletionRequiresQuotes in QuoteCompletionText?

Reply:
It looks like duplicate but it protects from regression if we will improve CompletionRequiresQuotes and change its tests.

Reply:
Yep makes sense, I have added tests in 5cb39e0a9981a7343e41a988720391dc10eb8d29